firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -213,9 +217,11 @@ String getName() {
   @Override
   public Task<String> getId() {
     preConditionChecks();
-    Task<String> task = addGetIdListener();
-    backgroundExecutor.execute(this::doGetId);
-    return task;
+    synchronized (lockGetFid) {
```

Comment:
I see that you want to emulate the old behavior of this being serialized on a sinle thread, but I don't know of adding an additional synchronization block is the right way to go.

For one thing, it isn't multi-process safe.

Reply:
Actually, everything doGetId() does it already thread and process safe, so this doesn't need its own protection, does it?

Reply:
I missed that local variables are thread safe. PTAL